1-Draft/RFCNNNN-ConvertFrom-Html.md

```diff
@@ -0,0 +1,185 @@
+```
+
+### PARAMETERS
+
+#### -InputObject
+
+Specifies the HTML strings to convert to HTML objects.
+Enter a variable that contains the string, or type a command or expression that gets the string.
+You can also pipe a string to **ConvertFrom-Html**.
+
+The *InputObject* parameter is required, but its value can be an empty string.
+When the input object is an empty string, **ConvertFrom-Html** does not generate any output.
+The *InputObject* value cannot be $Null.
+
+```yaml
+Type: String
+Parameter Sets: (All)
+Aliases:
+
+Required: True
+Position: 0
+Default value: None
+Accept pipeline input: True (ByValue)
+Accept wildcard characters: False
+```
+
+#### CommonParameters
+
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see about_CommonParameters (http://go.microsoft.com/fwlink/?LinkID=113216).
+
+### INPUTS
+
+#### System.String
+
+You can pipe a HTML string to **ConvertFrom-Html**.
+
+### OUTPUTS
+
+#### PSCustomObject
+
+### Examples
+
+You can provide a string to convert to a Html object using the `InputObject` Parameter
+
+```powershell
+ConvertFrom-Html -InputObject $InvokeWebRequestObject
+```
+
+You can provide a string to convert to a Html object using Position 0 (`InputObject`) parameterization:
+
+```powershell
+ConvertFrom-Html $InvokeWebRequestObject
+```
+
+You can pipe a string into `ConvertFrom-Html`:
+
+```powershell
+$htmlString = @"
+<HTML>
+
+<HEAD>
+    <TITLE>Your Title Here</TITLE>
+</HEAD>
+
+<BODY BGCOLOR="FFFFFF">
+    <CENTER>
+        <IMG SRC="clouds.jpg" ALIGN="BOTTOM"> </CENTER>
+    <HR>
+    <a href="http://somegreatsite.com">Link Name</a>
+    is a link to another nifty site
+    <H1>This is a Header</H1>
+    <H2>This is a Medium Header</H2>
+    Send me mail at
+    <a href="mailto:support@yourcompany.com">
+        support@yourcompany.com</a>.
+    <P> This is a new paragraph!
+        <P>
+            <B>This is a new paragraph!</B>
+            <BR>
+            <B>
+                <I>This is a new sentence without a paragraph break, in bold italics.</I>
+            </B>
+            <HR>
+</BODY>
+
+</HTML>
+"@
+
+$htmlObject = $htmlString | ConvertFrom-Html
+```
+
+Advanced example using Invoke-WebRequest and converting the returned content to a PSCustomObject.
+
+```powershell
+$dnsDumpsterURL = 'https://dnsdumpster.com/'
+$dumpsterRequest = Invoke-WebRequest -Uri $dnsDumpsterURL -SessionVariable session
+
+$props = @{
+    Uri        = $dnsDumpsterURL
+    Headers = @{Referer = $dnsDumpsterURL; 'Content-Type' = 'application / x-www-form-urlencoded'}
+    WebSession = $session
+    Body       = @{
+      'csrfmiddlewaretoken' = $dumpsterRequest.InputFields.value;
+      'targetip' = 'microsoft.com'
+    }
+    Method     = 'Post'
+}
+
+$dnsDumpsterObject = (Invoke-WebRequest @props).Content | ConvertFrom-Html
```

Comment:
Acturally, this will work:

```powershell
$dnsDumpsterObject = Invoke-WebRequest @props | ConvertFrom-Html
```

Paramater binding will convert `BasicHtmlWebResponseObject` to a string which will be the contents of `Content`.

Reply:
@markekraus  Thanks for this, and this makes sense.  I'll update the RFC.